doc/man3/d2i_X509.pod

```diff
@@ -538,6 +540,22 @@ Alternative technique:
  if (d2i_X509(&x, &p, len) == NULL)
      /* error */
 
+Setting up a library context and property query:
+
+ X509 *x;
+ unsigned char *buf;
+ const unsigned char *p;
+ int len;
+ OSSL_LIB_CTX *libctx = ....;
+ const char *propq = ....;
+
+ /* Set up buf and len to point to the input buffer. */
+ p = buf;
+ x = X509_new_ex(libctx, propq);
+
+ if (d2i_X509(&x, &p, len) == NULL)
+     /* error, x is freed and NULL assigned to it */
```

Comment:
The phrase `x is freed` is unclear. Is it **already** freed inside `d2_X509`, or is the programmer *expected* to free it in the error path?

Reply:
changing to past tense makes it hopefully clear